src/main/java/com/microsoft/azure/vmagent/AzureVMManagementServiceDelegate.java

```diff
@@ -1561,21 +1559,27 @@ public static String verifyStorageAccountName(
         boolean isAvailable = false;
         try {
             Azure azureClient = TokenCache.getInstance(servicePrincipal).getAzureClient();
-            //if it's not available we need to check if it's already in our resource group
-            isAvailable = azureClient.storageAccounts().checkNameAvailability(storageAccountName).isAvailable();
-            StorageAccount checkAccount = azureClient.storageAccounts().getByGroup(resourceGroupName, storageAccountName);
-            if ( !isAvailable && null == checkAccount ) {
+
+            CheckNameAvailabilityResult checkResult = azureClient.storageAccounts().checkNameAvailability(storageAccountName);
+            isAvailable = checkResult.isAvailable();
+            if( !isAvailable && checkResult.reason().equals(Reason.ACCOUNT_NAME_INVALID)) {
+                return Messages.Azure_GC_Template_SA_Cant_Validate();
```

Comment:
The message is "Could not validate the storage account name.", which doesn't make sense in this case. Better to be something like "the storage account name is not valid, a valid name can contain only lowercase letters and numbers, and must between 3 and 24 characters" (I copied this from azure portal)